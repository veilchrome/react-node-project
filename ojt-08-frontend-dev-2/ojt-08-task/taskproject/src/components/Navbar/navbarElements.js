// Element untuk navbar
// Referensi : https://www.geeksforgeeks.org/create-a-responsive-navbar-using-reactjs/
import { FaBars } from "react-icons/fa";
import { NavLink as Link } from "react-router-dom";
import styled from "styled-components";

export const Nav = styled.nav`
  background: #1a1a1a;
  font-family: "Poppins", sans-serif;
  font-weight: 500;
  height: 85px;
  display: flex;
  justify-content: space-between;
  padding: 0.2rem calc((100vw - 1000px) / 2);
  z-index: 12;
`;

export const NavLink = styled(Link)`
  color: #808080;
  display: flex;
  align-items: center;
  text-decoration: none;
  padding: 0 1rem;
  height: 100%;
  cursor: pointer;
  &.active {
    color: #ddd;
  }
`;

export const Bars = styled(FaBars)`
  display: none;
  color: #808080;
  @media screen and (max-width: 768px) {
    display: block;
    position: absolute;
    top: 0;
    right: 0;
    transform: translate(-100%, 75%);
    font-size: 1.8rem;
    cursor: pointer;
  }
`;

export const NavMenu = styled.div`
  display: flex;
  align-items: center;
  margin-right: -24px;
  white-space: nowrap;
  @media screen and (max-width: 768px) {
    display: none;
  }
`;
