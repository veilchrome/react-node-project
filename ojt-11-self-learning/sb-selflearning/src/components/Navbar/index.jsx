import { useState } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { FaMoon, FaSun } from "react-icons/fa";
import { NavLink } from "react-router-dom";

const NavContainer = styled.div`
  font-family: "Noto Sans", sans-serif;
  font-weight: 600;
  background-color: ${(props) => (props.isDarkMode ? "#0C2D57" : "#f4f4f4")};
  display: flex;
  align-items: center;
`;

const Button = styled.button`
  padding: 10px 20px;
  margin-right: 15px;
  margin-left: 10px;
  border: none;
  border-radius: 5px;
  background-color: ${(props) => (props.isDarkMode ? "#0B60B0" : "#ccc")};
  color: ${(props) => (props.isDarkMode ? "#fff" : "#333")};
  cursor: pointer;
  transition: background-color 0.3s ease, color 0.3s ease;

  &:hover {
    background-color: ${(props) => (props.isDarkMode ? "#777" : "#aaa")};
  }

  &:focus {
    outline: none;
  }
`;

const Nav = styled.nav`
  ul {
    list-style-type: none;
    padding: 20px;
    margin: 0;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  li {
    padding: 10px 20px;
    cursor: pointer;
    transition: background-color 0.3s ease, border-radius 0.3s ease;
    border-radius: 10px;
    text-decoration: none;

    /* Custom styles to remove blue color of active link */
    & > a {
      color: ${(props) => (props.isDarkMode ? "#fff" : "#333")};
      text-decoration: none;
    }

    &:hover {
      background-color: ${(props) => (props.isDarkMode ? "#555" : "#ccc")};
      border-radius: 20px;
    }
  }
`;

const Navbar = ({ isDarkMode }) => {
  const [darkMode, setDarkMode] = useState(isDarkMode);

  const toggleMode = () => {
    setDarkMode(!darkMode);
  };

  return (
    <NavContainer isDarkMode={darkMode}>
      <Button onClick={toggleMode} isDarkMode={darkMode}>
        {darkMode ? <FaSun /> : <FaMoon />}
      </Button>
      <Nav isDarkMode={darkMode}>
        <ul>
          <li>
            <NavLink to="/news">News</NavLink>
          </li>
          <li>
            <NavLink to="/opinion">Opinion</NavLink>
          </li>
          <li>
            <NavLink to="/sport">Sport</NavLink>
          </li>
          <li>
            <NavLink to="/culture">Culture</NavLink>
          </li>
          <li>
            <NavLink to="/lifestyle">Lifestyle</NavLink>
          </li>
        </ul>
      </Nav>
    </NavContainer>
  );
};

Navbar.propTypes = {
  isDarkMode: PropTypes.bool.isRequired,
};

export default Navbar;
