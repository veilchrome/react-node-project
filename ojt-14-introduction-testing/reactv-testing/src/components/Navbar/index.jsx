const Navbar = () => {
  return (
    <div className="Navbar" data-testid="navbar">
      <nav>
        <ul>
          <li>Gandum A</li>
          <li>Gandum C</li>
          <li>Gandum B</li>
        </ul>
      </nav>
    </div>
  );
};

export default Navbar;
