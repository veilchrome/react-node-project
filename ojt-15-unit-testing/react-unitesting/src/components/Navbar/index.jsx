import styled from "styled-components";

const Navbar = () => {
  const NavbarStyle = styled.nav`
    background-color: #333;
    padding: 10px 20px;
    ul {
      list-style-type: none;
      margin: 0;
      padding: 0;
      display: flex;
      justify-content: space-around;
    }
    li {
      display: inline;
      margin-right: 10px;
      color: #fff;
      font-size: 18px;
      cursor: pointer;
    }
    li:hover {
      text-decoration: underline;
    }
  `;

  return (
    <div className="Navbar" data-testid="navbar">
      <NavbarStyle>
        <ul>
          <li>Gandum A</li>
          <li>Gandum C</li>
          <li>Gandum B</li>
        </ul>
      </NavbarStyle>
    </div>
  );
};

export default Navbar;
