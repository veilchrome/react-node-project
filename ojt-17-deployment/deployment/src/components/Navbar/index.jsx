import React from "react";
import styled from "styled-components";
import { NavLink } from "react-router-dom";

const Navbar = () => {
  const NavbarStyle = styled.nav`
    background-color: #333;
    padding: 10px 20px;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);

    ul {
      list-style-type: none;
      margin: 0;
      padding: 0;
      display: flex;
    }
    li {
      display: inline;
      margin: 10px;
      font-size: 18px;
      cursor: pointer;
    }
    li a {
      color: #fff;
      text-decoration: none;
    }
    li a:hover {
      text-decoration: underline;
    }
  `;

  const CustomNavLink = styled(NavLink)`
    color: #fff;
    text-decoration: none;
    &:hover {
      text-decoration: underline;
    }
  `;

  return (
    <div className="Navbar" data-testid="navbar">
      <NavbarStyle>
        <ul>
          <li>
            <CustomNavLink to="/">Home</CustomNavLink>
          </li>
          <li>
            <CustomNavLink to="/registration">Registration</CustomNavLink>
          </li>
          <li>
            <CustomNavLink to="/anilist">Anilist</CustomNavLink>
          </li>
          <li>
            <CustomNavLink to="/todos">Todo List</CustomNavLink>
          </li>
        </ul>
      </NavbarStyle>
    </div>
  );
};

export default Navbar;
