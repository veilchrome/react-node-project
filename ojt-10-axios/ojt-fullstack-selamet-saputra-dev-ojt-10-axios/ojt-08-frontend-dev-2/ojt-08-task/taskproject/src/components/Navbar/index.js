import React from "react";
import { Nav, Bars, NavLink, NavMenu } from "./navbarElements";

const Navbar = () => {
  return (
    <>
      <Nav>
        <Bars />
        <NavMenu>
          <NavLink to="/"></NavLink>
          <NavLink to="/about">About</NavLink>
          <NavLink to="/contact">Contact</NavLink>
          <NavLink to="/gallery">Gallery</NavLink>
          <NavLink to="/hook">Hook</NavLink>
        </NavMenu>
      </Nav>
    </>
  );
};

export default Navbar;
