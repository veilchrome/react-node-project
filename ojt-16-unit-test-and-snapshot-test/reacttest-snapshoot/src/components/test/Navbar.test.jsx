import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Navbar from "../Navbar";

test("renders navbar with correct links", () => {
  const { container } = render(
    <MemoryRouter>
      <Navbar />
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});
