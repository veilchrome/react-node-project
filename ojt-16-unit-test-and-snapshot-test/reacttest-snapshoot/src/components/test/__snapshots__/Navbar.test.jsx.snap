// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders navbar with correct links 1`] = `
<div>
  <div
    class="Navbar"
    data-testid="navbar"
  >
    <nav
      class="sc-beySPh cReCLt"
    >
      <ul>
        <li>
          <a
            aria-current="page"
            class="sc-guDLey clbyy active"
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            class="sc-guDLey clbyy"
            href="/registration"
          >
            Registration
          </a>
        </li>
        <li>
          <a
            class="sc-guDLey clbyy"
            href="/anilist"
          >
            Anilist
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
